src/caselawclient/models/identifiers/neutral_citation.py

```diff
@@ -0,0 +1,35 @@
+import re
+
+from . import Identifier, IdentifierSchema
+
+VALID_NCN_PATTERN = re.compile(r"(^\[([0-9]{4})\] ([a-zA-Z]+)(?: ([a-zA-Z]+))? ([0-9]+)(?: \(([a-zA-Z]+)\))?$)")
```

Comment:
I think I'd like named capture groups

Reply:
I don't disagree, but it's a bit of extra work we don't need at the moment. I've left some notes that when we move all the NCN-based URL generation code into this class that we should tidy this up with named groups.